### PR TITLE
adds ability for users to delete own comments

### PR DESCRIPTION
**Pull request by addleton**

### CODE REVIEW

**Review by mezzDavies — CHANGES_REQUESTED** [review]
This looks good but there a few things we need to consider for full functionality:

We should make the api delete request and then, if successful, re-render the comments with the deleted comment removed. No need to refetch the comments though.

If the delete request fails (eg network error) we should feedback to the user that the deletion of their comment hasn't worked.

Also some way of feeding back that the deletion is in progress (similar to loading) is required.

File: src/utils/api.js [hunk 1]
```diff
@@ -28,3 +28,7 @@ export const postCommentByArticle = (articleId, user, input) => {
   };
   return api.post(`/articles/${articleId}/comments`, comment);
 };
+
+export const deleteSingleUserComment = (commentId) => {
+  return api.delete(`/comments/${commentId}`);
```
Comment (Harrison365) [inline src/utils/api.js:33]: Nice 👍

File: src/components/SingleComment.jsx [hunk 2]
```diff
@@ -5,9 +5,33 @@ import Chip from "@mui/joy/Chip";
...
+
+  const handleDelete = () => {
+    if (isDeleted) {
+      return;
+    }
+    setIsDeleted(true);
+    deleteSingleUserComment(comment.comment_id)
+      .then(() => {
+        setComments((currentComments) => {
+          return currentComments.slice(1);
```
Comment (Harrison365) [inline src/components/SingleComment.jsx]: This appears to delete the first comment not the selected comment :(

Comment (Harrison365) [reply src/components/SingleComment.jsx]: You will need to use a filter to remove the specific comment deleted.

Comment (addleton) [reply src/components/SingleComment.jsx]: On it now! Not sure how I missed that...

**Review by itskatherine — COMMENTED** [review]
Smashed it Michael, good to merge! 👍 🎉

**Pull request merged**